internal/notif/elasticsearch/client.go

```diff
@@ -0,0 +1,137 @@
+package elasticsearch
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/crazy-max/diun/v4/internal/model"
+	"github.com/crazy-max/diun/v4/internal/msg"
+	"github.com/crazy-max/diun/v4/internal/notif/notifier"
+	"github.com/crazy-max/diun/v4/pkg/utl"
+	"github.com/pkg/errors"
+)
+
+// Client represents an active elasticsearch notification object
+type Client struct {
+	*notifier.Notifier
+	cfg  *model.NotifElasticsearch
+	meta model.Meta
+}
+
+// New creates a new elasticsearch notification instance
+func New(config *model.NotifElasticsearch, meta model.Meta) notifier.Notifier {
+	return notifier.Notifier{
+		Handler: &Client{
+			cfg:  config,
+			meta: meta,
+		},
+	}
+}
+
+// Name returns notifier's name
+func (c *Client) Name() string {
+	return "elasticsearch"
+}
+
+// Send creates and sends an elasticsearch notification with an entry
+func (c *Client) Send(entry model.NotifEntry) error {
+	username, err := utl.GetSecret(c.cfg.Username, c.cfg.UsernameFile)
+	if err != nil {
+		return err
+	}
+
+	password, err := utl.GetSecret(c.cfg.Password, c.cfg.PasswordFile)
+	if err != nil {
+		return err
+	}
+
+	// Use the same JSON structure as webhook notifier
+	message, err := msg.New(msg.Options{
+		Meta:  c.meta,
+		Entry: entry,
+	})
+	if err != nil {
+		return err
+	}
+
+	body, err := message.RenderJSON()
+	if err != nil {
+		return err
+	}
+
+	// Parse the JSON to add the client field
+	var doc map[string]any
+	if err := json.Unmarshal(body, &doc); err != nil {
+		return err
+	}
+
+	// Add the current time
+	doc["@timestamp"] = time.Now().Format(time.RFC3339Nano)
+
+	// Add the client field from the configuration
+	doc["client"] = c.cfg.Client
+
+	// Re-marshal the JSON with the client field
+	body, err = json.Marshal(doc)
+	if err != nil {
+		return err
+	}
+
+	// Build the Elasticsearch indexing URL
+	// This uses the Index API (POST /{index}/_doc) to create a document with an auto-generated _id:
+	// https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-create
+	url := fmt.Sprintf("%s://%s:%d/%s/_doc", c.cfg.Scheme, c.cfg.Host, c.cfg.Port, c.cfg.Index)
```

Comment:
I looked a bit more at the docs and wonder if we could use their go client instead? https://github.com/elastic/go-elasticsearch

Also I think we should remove `Scheme`, `Host` and `Port` and replace them with an `Address` one that makes more sense looking at the API docs https://pkg.go.dev/github.com/elastic/go-elasticsearch#pkg-overview

So instead of:

```yaml
  elasticsearch:
    scheme: https
    host: localhost
    port: 9200
```

It would be:

```yaml
  elasticsearch:
    address: https://localhost:9200
```

Seems also they support multiple addresses, maybe we should too?

Reply:
These are good points. I initially oriented myself at the mqtt and webhook implementation and tried to stay as close to them as possible to have uniform code, but it makes sense to implement it properly with their client. I'll look into it, thanks for your input.

Reply:
I'd have to test, how forgiving their client is regarding elasticsearch version compatibility. The `_doc` endpoint is probably the most essential and basic endpoint, generally unchanged since version 5. So by using their go client, hopefully they don't restrict version compatibility, which would make it harder to maitain the integration. 

I'll come with some tests, probably by the end of next week.

Reply:
Thanks, I pushed an extra commit to at least switch to `Address` field and also validate it using url package.